first successfull implementation of the diff

AntonioSouls opened the pull request:
I've finally modified diff.py code and it works successfully. I'm waiting for next task!! Please check that everything works

Commit:
Added the diffparser class

Commit:
Added content of .gitignore

Commit:
Control Commit

Commit:
Added a parsing in the diff.py class (it will need to be moved to the parser class) and fixed up because doesn't work well

Commit:
Deleted the parser class and modified the diff.py. From now, we make diff simply comparing all the new file's lines with all the old file's lines and saving only differences

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
         self.file_old = file_old
+        self.file_diff = file_diff
         self.lista_file_new = None
         self.lista_file_old = None
```
use english name for the variables

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
+    def diff_function(self):
+        file_object_new = open(self.file_new, 'r')                             # Opening the new file for reading
+        file_object_old = open(self.file_old, 'r')                             # Opening the old file for reading
+        self.lista_file_new = file_object_new.readlines()                      # Saving the new file's lines into a list
```
same as before

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
-        # #                 self.lista_file_old.insert(diff_result.index(x), self.lista_file_new[diff_result.index(x)])
+    # Doing diff between two files: an Old File and a New File and saving into a third file only the differences between the preavious files
+    def diff_function(self):
+        file_object_new = open(self.file_new, 'r')                             # Opening the new file for reading
```
in python you can use the `with open(FILENAME, MODE) as f_name: ...` (https://stackoverflow.com/questions/9282967/how-to-open-a-file-using-the-open-with-statement)

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
-        return                                                  
\ No newline at end of file
+                                                                              # Starting the research of the differences between two files
+        for stringa_new in self.lista_file_new:
```
use english names

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
\ No newline at end of file
+                                                                              # Starting the research of the differences between two files
+        for stringa_new in self.lista_file_new:
+            esiste: bool = False
```
same

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
+                if stringa_new == self.lista_file_old[i] :
+                    esiste = True
+                i=i+1
+            if(esiste==False):
```
can be -> `if not esiste:` or `if not exist:` after the renaming

samirsalman commented on memory_diff/diff.py:
```diff
@@ -3,46 +3,38 @@
+        file_object_diff = open(self.file_diff, 'w')
+        file_object_diff.writelines(self.lista_file_diff)
+        file_object_diff.close()
+        return                                               
```
remove the return if you're returning `None`

samirsalman commented on test_memory_diff/Matecat_New.tmx:
```diff
@@ -6,10 +6,10 @@
       <prop type="matecat-match">MT</prop>
       <tuv xml:lang="en-GB">
-        <seg>Important legal notice, really important</seg>
+	  <seg>Importante avviso legale, veramente importante</seg>
```
remove the tmx files from github and add them to the .gitignore, both old, new and diff file

samirsalman commented on test_memory_diff/test_diff.py:
```diff
@@ -5,8 +5,9 @@
     def test_saving_diff(self):
-        diff_object = Diff('/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/fileNew.txt', '/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/fileOld.txt')
-        modified_file = diff_object.saving_diff()
+        diff_object = Diff('/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/Matecat_New.tmx', '/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/Matecat_Old.tmx','/mnt/c/Users/Antonio/Desktop/memory-diff/test_memory_diff/diff_file.tmx')
```
put the path as input of the test function, is not a best practice to hardcode paths

AntonioSouls replied:
I don't understand what i've to do

samirsalman replied:
instead of put the file path inside the code you should take the file path as input

AntonioSouls committed:
Delete Matecat_New.tmx

AntonioSouls committed:
Delete diff_file.tmx

AntonioSouls committed:
Delete Matecat_Old.tmx

AntonioSouls committed:
Delete fileNew.txt

AntonioSouls committed:
Delete fileOld.txt

Commit:
Final english translation and correction of reviews

Commit:
<Merge branch 'feature/parser' of https://github.com/AntonioSouls/memory-diff into feature/parser
This merge is necessary because i've removed the .tmx file from GitHub and, in the local repository, i've added them into .gitignore and after i've modified code into local repository. So I'm making this merge because I want to
re-align the remote repo with the local repo

Commit:
Completed English translation and fixed all the reviews

Commit:
Temporarily removed .tmx files to modifie the .gitignore

Commit:
Re-added .tmx files after editing .gitignore

AntonioSouls committed:
Delete Matecat_New.tmx

AntonioSouls committed:
Delete Matecat_Old.tmx

AntonioSouls committed:
Delete diff_file.tmx

Commit:
permanently changed .gitignore

Commit:
Changed the way to read .tmx files, this way is more clean

Commit:
Undo the changes of the previous commit because they were made on the wrong branch
